authors/apps/articles/views.py

```diff
@@ -47,6 +50,86 @@ def delete(self, request, slug):
         super().delete(self, request, slug)
         return Response({"message": "Article Deleted Successfully"})
 
+
+class RatingDetails(GenericAPIView):
+    queryset = Rating.objects.all()
+    serializer_class = RatingSerializer
+    permission_classes = (IsAuthenticatedOrReadOnly, IsOwnerOrReadonly)
+    renderer_classes = (RatingJSONRenderer,)
+
+    def get_rating(self, user, article):
+        """
+        Returns a rating given the user id and the article id
+        """
+        try:
+            return Rating.objects.get(user=user, article=article)
+        except Rating.DoesNotExist:
+            raise NotFound(detail={'rating': 'Rating not found'})
+
+    def get_article(self, slug):
```

Comment:
Hey @codingedward, I went over the PR and I believe this method here is not necessary since the same has been checked in line 134 and that has already been merged, you could take up the same.

Reply:
After implementing the same, kindly include a test for rating a non-existent article...seems crazy but I think its necessary.

Reply:
@Ramon3162, my bad, I seem to have moved the `get_article` method from the `ArticlesDetails`. I will fix this, thank you.

Reply:
I will most definitely add that test.